rotkehlchen/chain/ethereum/trades.py

```diff
@@ -0,0 +1,245 @@
+from __future__ import annotations
+
+from typing import (
+    Any,
+    Dict,
+    NamedTuple,
+    Tuple,
+    Union,
+)
+
+from rotkehlchen.assets.asset import EthereumToken
+from rotkehlchen.assets.unknown_asset import (
+    FakeAsset,
+    UnknownEthereumToken,
+)
+from rotkehlchen.serialization.deserialize import (
+    deserialize_asset_amount,
+    deserialize_ethereum_address,
+    deserialize_ethereum_token_from_db,
+    deserialize_fee,
+    deserialize_location_from_db,
+    deserialize_price,
+    deserialize_timestamp,
+    deserialize_trade_type_from_db,
+    deserialize_unknown_ethereum_token_from_db,
+)
+from rotkehlchen.typing import (
+    AssetAmount,
+    ChecksumEthAddress,
+    Fee,
+    Location,
+    Price,
+    Timestamp,
+    TradeType,
+)
+
+# Quick lookup for AMMTrade locations
+AMMTradeLocations = {Location.UNISWAP}
+AMMTradeDBTuple = (
+    Tuple[
+        str,  # tx_hash
+        int,  # log_index
+        str,  # address
+        int,  # timestamp
+        str,  # location
+        str,  # type
+        int,  # is_base_asset_unknown
+        str,  # base_asset_address
+        str,  # base_asset_symbol
+        str,  # base_asset_name
+        int,  # base_asset_decimals
+        int,  # is_quote_asset_unknown
+        str,  # quote_asset_address
+        str,  # quote_asset_symbol
+        str,  # quote_asset_name
+        int,  # quote_asset_decimals
+        str,  # amount
+        str,  # rate
+        str,  # fee
+        str,  # notes
+    ]
+)
+
+
+class AMMTrade(NamedTuple):
+    """This class aims for a better trades support than the current Trade class
+    in AMMs protocols.
+    """
+    tx_hash: str
+    log_index: int
+    address: ChecksumEthAddress
+    timestamp: Timestamp
+    location: Location
+    trade_type: TradeType
+    base_asset: Union[EthereumToken, UnknownEthereumToken]
+    quote_asset: Union[EthereumToken, UnknownEthereumToken]
+    amount: AssetAmount
+    rate: Price
+    fee: Fee
+    notes: str = ''
+
+    @classmethod
+    def get_amm_trade_from_db_tuple(
```

Comment:
for these kind of functions use the existing naming convention. We use "deserialize_from_db" for this